pallets/on-slash-vesting/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod test;
+
+use frame_support::{
+	sp_runtime::{traits::Convert, FixedPointNumber, FixedU128},
+	traits::{Currency, OriginTrait},
+};
+use pallet_vesting::Vesting;
+use sp_runtime::{traits::BlockNumberProvider, BoundedVec};
+
+pub trait OnSlash<AccountId, Balance: Clone> {
+	fn on_slash(account: &AccountId, amount: Balance);
+}
```

Comment:
as specified in the "Anything Else" of the PR description:
> For now the trait for slashing needs to be in the same crate we implement it since we can't impl a foreign trait on a foreign crate.